forge-gui/res/cardsfolder/upcoming/court_of_ardenvale.txt

```diff
@@ -0,0 +1,11 @@
+Name:Court of Ardenvale
+ManaCost:2 W W
+Types:Enchantment
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigMonarch | TriggerDescription$ When CARDNAME enters the battlefield, you become the monarch.
+SVar:TrigMonarch:DB$ BecomeMonarch | Defined$ You
+T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigBranch | TriggerDescription$ At the beginning of your upkeep, return target permanent card with mana value 3 or less from your graveyard to your hand. If you're the monarch, return that permanent card to the battlefield instead.
+SVar:TrigBranch:DB$ Branch | BranchConditionSVar$ X | BranchConditionSVarCompare$ GE1 | TrueSubAbility$ TrigPlay | FalseSubAbility$ TrigReturn
+SVar:TrigReturn:DB$ ChangeZone | ValidTgts$ Permanent.YouCtrl+cmcLE3 | TgtPrompt$ Select target permanent card with mana value 3 in your graveyard | Origin$ Graveyard | Destination$ Hand

```

Comment:
engine doesn't see these Tgts on condition subs
you need to move it to Branch effect
(AI might not like it)

Reply:
is there an existing card/s with example like those you mentioned?

Reply:
target only works on the main sub ability chain

additional sub abilities can't have target

to make it work, you need a Pump before the Branch / or Target on Branch (depending on AI?)

Reply:
@Hanmac @tool4ever  I update it works on player but not on AI?

Reply:
might need AI logic, but that can be done later

Reply:
@kevlahnota do you want to try to write it as condition instead?

Reply:
> @kevlahnota do you want to try to write it as condition instead?

done seems to work fine